src/aggregationinfo.cpp

```diff
@@ -394,15 +394,13 @@ AggregationInfo AggregationInfo::SecureMergeInfos(
 void AggregationInfo::Clear() {
     sortedMessageHashes.clear();
     sortedPubKeys.clear();
-    std::vector<uint8_t*> mapKeys;
-    for (auto &mapEntry : tree) {
-        delete[] mapEntry.second;
-        mapKeys.push_back(mapEntry.first);
-    }
-    for (const uint8_t* mapKey : mapKeys) {
-        delete[] mapKey;
+    if (!(tree.empty())) {
+        for (auto &mapEntry : tree) {
+            delete[] mapEntry.first;
+            delete[] mapEntry.second;
```

Comment:
Are you sure it's safe to delete these bytes while iterating through the map? I can't remember why I did it the original way

Reply:
I wouldn't call myself an expert on whether or not it's safe, but I have no idea why it wouldn't be. 

You had a loop that deleted the second, and pushed the first on to a vector, defined inside that scope and not used elsewhere. Then you had a loop that deleted everything in the vector. This combined the two loops and removed the vector.

If you needed to delete the second before the first due to a dependence, you could just reorder those two lines.

The loop I submitted still passes the tests and @jonspock agreed they should be equivalent.

Reply:
OK, we can merge it, since automated tests run valgrind and found no issues.